token-swap/program/src/instruction.rs

```diff
@@ -127,6 +151,33 @@ pub enum SwapInstruction {
     ///   8. `[writable]` Fee account, to receive withdrawal fees
     ///   9. '[]` Token program id
     Withdraw(Withdraw),
+
+    ///   Deposit some tokens into the pool.  The output is a "pool" token
+    ///   representing ownership into the pool. Input token is converted at
+    ///   the current ratio.
+    ///
+    ///   0. `[]` Token-swap
+    ///   1. `[]` $authority
+    ///   2. `[writable]` token_(A|B) SOURCE Account, amount is transferable by $authority,
+    ///   3. `[writable]` token_a Swap Account, may deposit INTO.
+    ///   4. `[writable]` token_b Swap Account, may deposit INTO.
+    ///   5. `[writable]` Pool MINT account, $authority is the owner.
+    ///   6. `[writable]` Pool Account to deposit the generated tokens, user is the owner.
+    ///   7. '[]` Token program id
+    DepositOneExactIn(DepositOneExactIn),
```

Comment:
I see what you mean... I wanted to make it clear that you're only depositing one token type instead of both, and that you're specifying exactly how much is going in, as opposed to the old deposit, which specifies the exact amount of pool tokens that you'll get out.  Since it's not actually performing a swap, `DepositAndSwap` seems misleading.  What about `DepositOneSideExactAmountIn` to be totally clear?

Reply:
Do you expect to also support depositing one side but specifying the input in pool tokens? If not, I don't think `ExactAmountIn` is needed. That can just be defined in the interface.
Are we still looking toward pools of >2 tokens?
How about `DepositSingleTokenType`

Reply:
It would be useful for people to be able to specify which side they want to fix when depositing or withdrawing, and it's something used in Balancer, so providing that info in this instruction would be best.  `DepositSingleTokenType` definitely clarifies the `One` part -- what about `DepositSingleTokenTypeExactAmountIn`?  For reference, I was looking at Balancer's terminology at https://github.com/balancer-labs/balancer-core/blob/master/contracts/BPool.sol -- but I found functions like `joinswapExternAmountIn` opaque, so I'm very motivated to make this clear for us!

Reply:
So hard to find something both precise and concise! ;)
`DepositSingleTokenTypeExactAmountIn` is at least very clear... wfm